README.md

```diff
@@ -23,117 +23,56 @@ axum-login
 
 ## 🎨 Overview
 
-`axum-login` is a Tower middleware providing session-based user authentication for `axum` applications.
-
-- Decouples user storage from authentication
-- Supports arbitrary user types and arbitrary storage backends
-- Provides methods for: logging in, logging out, and accessing current user
-- Optional role-based access controls via an arbitrary role type
-- Wraps [`axum-sessions`](https://github.com/maxcountryman/axum-sessions) to provide flexible sessions
-- Leverages `tower_http::auth::RequireAuthorizationLayer` to protect routes
-
-> **Note** `axum-login` implements a fundamental pattern for user authentication, however some features may be missing. Folks are encouraged to make suggestions for extensions to the library.
+This crate provides user identification, authentication, and authorization
+as a `tower` middleware for `axum`.
+
+If offers:
+
+- **User Identification, Authentication, and Authorization**: Leverage
+  `AuthSession` to easily manage authentication and authorization. This is
+  also an extractor, so it can be used directly in your `axum` handlers.
+- **Support for Arbitrary Users and Backends**: Applications implement a
+  couple of traits, `AuthUser` and `AuthnBackend`, allowing for any user
+  type and any user management backend. Your database? Yep. LDAP? Sure. An
+  auth provider? You bet.
+- **User and Group Permissions**: Authorization is supported via the
+  `AuthzBackend` trait, which allows applications to define custom
+  permissions. Both user and group permissions are supported.
+- **Convenient Route Protection**: Middleware for protecting access to
+  routes are provided via the `login_required` and `permission_required`
+  macros. Or bring your own by using `AuthSession` directly with
+  `from_fn`.
+- **Rock-solid Session Management**: Uses `tower-sessions`
+  for high-performing and ergonomic session management. _Look ma, no deadlocks!_
 
 ## 📦 Install
 
 To use the crate in your project, add the following to your `Cargo.toml` file:
 
 ```toml
 [dependencies]
-axum-login = "0.6.0"
+axum-login = "0.7.0"
```

Comment:
Imho it's a major version change. Maybe 1.0.0-rc?

Reply:
I'm comfortable moving to `1.0.0` with this API. However, I also anticipate we'll iterate a bit more on this before it's completely settled. Following [this semantic versioning spec](https://semver.org/#spec-item-4), I think we have more flexibility at `0.x.y` so my thinking is to wait a little longer.